### ISSUE TITLE
Full name is imported instead of Username

### ISSUE THREAD

**Issue by cboulanger** [ConnectedCommunities/humhub-modules-bulk_import#4]
Thank you for fixing #3. I have tested it and passwords are now correctly imported. However, another problem remains: the username column is not imported, instead, the full name ("firstname lastname") is used as the user name. (See screenshot below).

![screenshot](https://cloud.githubusercontent.com/assets/75390/17513292/6a75ead6-5e2c-11e6-9537-483274553d44.PNG)

**Comment by boppy** [comment]
I've had the same problem with user data saved from Excel.

It's the encoding excel uses for CSV-Export. It's UTF-8-BOM. It adds these 3 bytes at the beginning of the CSV. You might check that with (I added it here: [MainController Line 201](https://github.com/ConnectedCommunities/humhub-modules-bulk_import/blob/master/controllers/MainController.php#L201))

```php
                    echo '<pre>'. print_r([
                            empty($data['username']),
                            array_keys($data)[0],
                            strlen(array_keys($data)[0]),
                            mb_strlen(array_keys($data)[0])
                        ], 1) .'</pre>';
```
It responded with `[true, 'username', 11, 9]`.

**My first solution** was to open the file with Notepad++ and change the encoding:

![encode-wo-bom](https://user-images.githubusercontent.com/749141/29915423-284e6e76-8e3c-11e7-932a-ba91a69659eb.png)

Another solution is be to remove the BOM after reading the CSV. After doing so I was able to import files from Excel without any converting. See https://github.com/boppy/humhub-modules-bulk_import/tree/boppy-patch-BOM - I'll provide a PR if you want to.

I'm sorry for the massive changes in `MainController.php` - but the file did not really follow any coding rule so my IDE reformatted it. Only the lines 181 and 182 of the controller files are actually changed.
